tests/e2e/test_app.py

```diff
@@ -95,7 +95,9 @@ def test_health(client):
 
 def test_upsert(client):
     # Upsert a document to the index
-    upsert_response = client.post("/context/upsert", json=upsert_payload.dict())
+    upsert_response = client.post(
+        f"/api/{API_VERSION}/context/upsert",
```

Comment:
That kinda misses the point... 

You want to pass a `base_url` arg to `TestClient()`, and leave the test cases themselves untouched.

Reply:
fixed